action_tests.py

```diff
@@ -0,0 +1,118 @@
+from unittest import TestCase
+
+from action import ContainerConfig, ContainerMount, prepare_dockerrun_file
+
+
+class MountsTest(TestCase):
+    @staticmethod
+    def _create_container_config(host_path: str, container_path: str):
+        return ContainerConfig(
+            dockerfile="docker/Dockerfile",
+            name="sample-application",
+            image_base_name="YourAwsAccountId.dkr.ecr.YourAwsRegion.amazonaws.com/your-ecr-repository",
+            ports=(),
+            mounts=(
+                ContainerMount(
+                    host_path=host_path,
+                    container_path=container_path,
+                    read_only=False,
+                ),
+            ),
+        )
+
+    def test_single_mount_conversion(self):
+        containers = (
+            self._create_container_config("/var/my-sample-application-test/data", "/opt/data"),
+        )
+
+        file = prepare_dockerrun_file(containers, "app-version")
+
+        self.assertDictEqual(
```

Comment:
Ich fände es gut, den Copy-Paste-Anteil zu verringern, entweder durch Auslagerung in gemeinsame Teile oder  durch Refactoring des Codes, so dass die Logik bzgl. der Mounts unabhängig vom ganzen Rest, wie z.B. den image names, memory reservations etc., getestet werden kann.

Reply:
Ich hätte mir auch ein Template-Struktur gewünscht, ansonsten werden die Tests ganz schnell unbeliebt, weil sehr mühsam anzupassen und schwierig zu warten...